core/src/subgraph/instance_manager.rs

```diff
@@ -606,10 +610,12 @@ where
 
                     debug!(
                         &logger,
-                        "Block stream produced a non-fatal error";
+                        "Block stream produced a non-fatal error. Retrying";
                         "error" => format!("{}", e),
                     );
-                    continue;
+
+                    // Exit inner block stream consumption loop and go up to loop that restarts subgraph
+                    break;
```

Comment:
This case I think should be a continue, the polling block stream at least is designed to be able to recover from its errors.

Reply:
You are right, `FirehoseBlockStream` will also reconnect forever in case of connection drop, I've reverted.